src/Response/HttpResponseProcessor/ErrorProcessor.php

```diff
@@ -35,13 +35,16 @@ protected function getPayload(
             // This handles inconsistency in ShipX API. Some errors have HTTP status code 200 and error data in body.
             if (isset($data['status']) && $data['status'] !== 200 && isset($data['key']) && isset($data['error'])) {
                 return new Error([
-                    'status' => $data['status'],
-                    'error' => $data['key'],
-                    'message' => $data['error']
+                    'status' => $data['status'] ?? $httpResponse->getStatusCode(),
+                    'error' => $data['key'] ?? '',
+                    'message' => $data['error'] ?? '',
                 ]);
             }
             return null;
         }
+        $data['status'] ??= $httpResponse->getStatusCode();
+        $data['error'] = $data['key'] ?? '';
+        $data['message'] = $data['error'] ?? '';
```

Comment:
@snapshotpl The default (documented) error response has `status`, `error`, `message` and (optional) `details`. Please fill in `message` with `error` and `error` with `key` only if `message` doesn't exist.

I know the whole thing is stupid, but inconsistency of Inpost API is even more stupid :-1:.